### PR TITLE
feature: WEOS-1134 As a user I should be able to filter a list of items for a content type

### PR DESCRIPTION
**Pull request by shaniah868**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This introduces the filter functionality for the list endpoint 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: controllers/rest/controller_standard_test.go [hunk 1]
```diff
@@ -1075,6 +1075,159 @@ func TestStandardControllers_List(t *testing.T) {
...
+		result := "invalid property found in filter: fgsd"
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Errorf("unexpected error : %s", err)
+		}
+		if strings.Contains(result, string(b)) {
+			t.Errorf("expected error returned to be %s got %s", result, string(b))
+		}
+	})
+	t.Run("Sending multiple values on the wrong operator in the generic list endpoint as parameters", func(t *testing.T) {
```
Comment (akeemphilbert) [inline controllers/rest/controller_standard_test.go:1232]: 2 more tests could get 100% coverage on the ListMiddleware, one that confirms entityFactory nil returns and error and the other is testing that setting page = 0 results in page being set to 1

File: projections/gorm.go [hunk 2]
```diff
@@ -282,10 +305,107 @@ func sort(order map[string]string) func(db *gorm.DB) *gorm.DB {
...
+func DateTimeCheck(entityFactory weos.EntityFactory, properties map[string]FilterProperty) (map[string]FilterProperty, error) {
+	var err error
+	schema := entityFactory.Schema()
+	for key, value := range properties {
+		if schema.Properties[key] != nil && schema.Properties[key].Value.Format == "date-time" {
+			_, err := time.Parse(time.RFC3339, value.Value)
+			if err != nil {
+				return nil, err
+			}
+			//count := 1
```
Comment (akeemphilbert) [inline projections/gorm.go]: remove if not used

File: projections/gorm.go [hunk 3]
```diff
@@ -282,10 +305,107 @@ func sort(order map[string]string) func(db *gorm.DB) *gorm.DB {
...
+			//}
+
+		}
+	}
+
+	return properties, err
+}
+
+//filterStringBuilder is used to build the query strings
+func filterStringBuilder(dbName string, properties map[string]FilterProperty) string {
```
Comment (akeemphilbert) [inline projections/gorm.go]: why are you building the query manually and not using form for or and and conditions https://gorm.io/docs/query.html#Or-Conditions (you will run into issues when we try to run this across database types). It might be cleaner to make a Gorm Scope where you pass in the filterOptions and then in there update the db query with the and options and checking the operator to set what goes in the where clause e.g. ` .Where("name LIKE ?", "group%")`

**Review by akeemphilbert — CHANGES_REQUESTED** [review]
remove the test.db

**Review by akeemphilbert — APPROVED** [review]

**Review by Aidan-E — APPROVED** [review]
Good job👍🏾
